app/src/main/java/cm/aptoide/pt/AptoideApplication.java

```diff
@@ -494,27 +494,27 @@ public AptoideDownloadManager getDownloadManager() {
     return downloadManager;
   }
 
-  public InstallManager getInstallManager(int installerType) {
+  public InstallManager getInstallManager() {
 
     if (installManagers == null) {
       installManagers = new SparseArray<>();
     }
 
-    InstallManager installManager = installManagers.get(installerType);
+    InstallManager installManager = installManagers.get(InstallerFactory.DEFAULT);
     if (installManager == null) {
 
       installManager = new InstallManager(getApplicationContext(), getDownloadManager(),
           new InstallerFactory(new MinimalAdMapper(),
               new InstallFabricEvents(analyticsManager, installAnalytics,
-                  getDefaultSharedPreferences(), rootAvailabilityManager),
-              getImageCachePath()).create(this, installerType), getRootAvailabilityManager(),
+                  getDefaultSharedPreferences(), rootAvailabilityManager)).create(this),
+          getRootAvailabilityManager(),
           getDefaultSharedPreferences(),
           SecurePreferencesImplementation.getInstance(getApplicationContext(),
               getDefaultSharedPreferences()),
           RepositoryFactory.getDownloadRepository(getApplicationContext().getApplicationContext()),
           RepositoryFactory.getInstalledRepository(
               getApplicationContext().getApplicationContext()));
-      installManagers.put(installerType, installManager);
+      installManagers.put(InstallerFactory.DEFAULT, installManager);
```

Comment:
You can remove the use of a SparseArray and only track the InstallManager instance.
This way you can remove the static string from InstallerFactory